### ISSUE TITLE
ASP.NET Core identity UI requests email but uses username

### ISSUE THREAD

**Issue by weitzhandler** [aspnet/AspNetCore#11786]
### Describe the bug
Hi folks,

I'm setting up the identity as follows:

```c#
services
        .AddDefaultIdentity<Tenant>()
        .AddDefaultUI(UIFramework.Bootstrap4)
        .AddEntityFrameworkStores<CatalogDbContext>();
```

Then in the `OnModelCreating` I'm initializing a user like this:
```c#
builder.Entity<Tenant>(t =>
  {
    var tenant = new Tenant { ... }
    var hasher = new PasswordHasher<Tenant>();
    tenant.PasswordHash = hasher.HashPassword(tenant, "mypass");
    t.HasData(tenant);
  });
```

When I go to the login page and try to login with the provided email/password, I get an error "Invalid login attempt".
Why is that?

I tried using a custom sign-in manager, and I see that the `public override Task<SignInResult> PasswordSignInAsync(string userName, string password, bool isPersistent, bool lockoutOnFailure)`  overload is called, but the argument for `userName` is the email. Typing in a username in the identity UI causes a client-side validation error.

### Expected behavior
The identity UI should ask for username instead of email

### Screenshots
![image](https://user-images.githubusercontent.com/2716316/60483250-51f01d80-9c9d-11e9-91bf-2a137becbd27.png)

![image](https://user-images.githubusercontent.com/2716316/60483289-7e0b9e80-9c9d-11e9-9aae-10dfed91f478.png)

### Additional context
P.S. I'm using IdentityServer in my project, but in a different scheme.

Include the output of `dotnet --info`: https://gist.github.com/weitzhandler/f3287c91722ec442d645b6320405c774

**Comment by weitzhandler** [comment]
I notice something weird:"

    public override async Task<SignInResult> PasswordSignInAsync(string userName, string password, bool isPersistent, bool lockoutOnFailure)
    {
      userName = userName.Split("@")[0];
      // val is assigned to the user I'm after
      var val = await UserManager.Users.SingleOrDefaultAsync(u => u.UserName == userName);
      // remains null!
      var val2 = await UserManager.FindByNameAsync(userName);

      if (val == null)
      {
        return SignInResult.Failed;
      }
      return await PasswordSignInAsync(val, password, isPersistent, lockoutOnFailure);
    }

**Comment by enetstudio** [comment]
The identity UI should ask for username instead of email

As far as I know, username and email are interchangeable. In other words, username is the email. The field username in the database is filled with the email data.

**Comment by blowdart** [comment]
By default in the templates username and password are synchronized. If you want to move away from this expectation you need to adjust the UI according to your needs.

**Comment by weitzhandler** [comment]
But it can't retrieve the user by neither email or username:

![image](https://user-images.githubusercontent.com/2716316/60533343-1179b980-9d08-11e9-87ca-594b0836a983.png)
 In the code example above, except for `val`, `byEmail` `val2` and `val3` are all null.

**Comment by weitzhandler** [comment]
Something ain't right, that's for sure. Either display username field, or use email for user retrieval.
You can't ask for an *email* and check the *username*  field for it, that makes no sense.

**Comment by blowdart** [comment]
We're away it's not ideal, and in 3..0 the templates get a little overhaul, so we consider if solved in 3.0
